### DISCUSSION TITLE
Reason why non-volatile storage partition sector size has to be < UINT16_T MAX?

### DISCUSSION THREAD

**Discussion by vChavezB** [zephyrproject-rtos/zephyr#60937] [Q&A] [answered]
See this condition

https://github.com/zephyrproject-rtos/zephyr/blob/488fd89e2bd64c60354378fdec33b7e65c713d38/subsys/settings/src/settings_nvs.c#L369C2-L369C2

I guess it has to do that the maximum value expected for `CONFIG_SPI_NOR_FLASH_LAYOUT_PAGE_SIZE` is 65536?

**Comment by Laczen** [comment]
This is because NVS is internally using a location descriptor that only supports blocks that are smaller then 65536. It is a NVS limitation that cannot be removed without heavily modifying nvs.

**Reply by vChavezB** [reply]
Could you point out where in the subsys this is implemented (the location descriptor you mentioned). It would be nice to know to understand more about the NVS.

Best regards and thanks for the info.

**Reply by Laczen** [reply, marked as answer]
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/fs/nvs/nvs_priv.h#L14-L22.

All `uint32_t *addr` used in the routines are formatted according to the above lines. It is known that nvs is unusable on devices with larger sector size (e.g. some stm32 devices).
